extensions/replication/tasks/EchoBucket.js

```diff
@@ -0,0 +1,203 @@
+        const canonicalId = sourceEntry.getOwnerCanonicalID();
+        let displayName;
+        let email;
+        let accountCreds;
+
+        const sourceS3Vault = this.vaultClientCache.getClient('source:s3');
+        const sourceAdminVault =
+                  this.vaultClientCache.getClient('source:admin');
+        async.waterfall([
+            done => {
+                // XXX HACK: will not work with > 1000 account, needs
+                // a proper route to get account name from canonical
+                // ID
+                sourceAdminVault.listAccounts(
+                    { maxItems: 1000 }, (err, data) => {
+                        if (err) {
+                            return done(err);
+                        }
+                        data.accounts.forEach(account => {
+                            if (account.canonicalId === canonicalId) {
+                                displayName = account.name;
+                            }
+                        });
+                        if (!displayName) {
+                            return done(errors.InternalError);
+                        }
+                        return done();
+                    });
+            },
+            done => sourceS3Vault.getEmailAddresses(
+                [canonicalId], { reqUid: log.getSerializedUids() }, done),
+            (res, done) => {
+                email = res.message.body[canonicalId];
+                accountCreds = this.accountCredsCache[canonicalId];
+                if (accountCreds) {
+                    return done(null, null);
+                }
+                return sourceAdminVault.generateAccountAccessKey(
+                    displayName, done);
+            }, (res, done) => {
+                if (res) {
+                    accountCreds = {
+                        accessKeyId: res.id,
+                        secretAccessKey: res.value,
+                    };
+                    this.accountCredsCache[canonicalId] = accountCreds;
+                }
+                return done();
+            },
+        ], err => {
+            if (err) {
+                return done(err);
+            }
+            return done(null, displayName, email, accountCreds);
+        });
+    }
+
+    _getTargetAccountCreds(sourceEntry, displayName, email, log, done) {
+        let canonicalId;
+        let accountCreds;
+        const { host, port } = this.destHosts.pickHost();
+        // if no nginx proxy is used, the client port is preset in the
+        // profile and the provided port is only used in the cache key
+        const destAdminVault = this.vaultClientCache.getClient('dest:admin',
+                                                               host, port);
+        const destS3Vault = this.vaultClientCache.getClient('dest:s3',
+                                                            host, port);
+
+        async.waterfall([
+            done => destS3Vault.getCanonicalIds(
```

Comment:
No, because Vault realms are different between source and destination, we need to fetch the destination canonical ID which is different than the source one, to set it properly in the destination metadata.